### PR TITLE
10.0 fix access apps allow remove

### PR DESCRIPTION
**Pull request by iledarn**

### CODE REVIEW

File: access_apps/models/res_config.py [hunk 1]
```diff
@@ -26,4 +26,42 @@ def default_get(self, fields):
...
+        for name, module in classified['to_uninstall']:
+            res[name] = module.state in ('installed', 'to install', 'to upgrade')
+            if self._fields[name].type == 'selection':
+                res[name] = int(res[name])
+
+        return res
+
+    @api.model
+    def _get_classified_fields(self):
+        # import wdb; wdb.set_trace()
```
Comment (yelizariev) [inline access_apps/models/res_config.py]: remove temporary line

File: access_apps/models/res_config.py [hunk 2]
```diff
@@ -26,4 +26,42 @@ def default_get(self, fields):
...
+        classified = self._get_classified_fields()
+        for name, module in classified['to_uninstall']:
+            res[name] = module.state in ('installed', 'to install', 'to upgrade')
+            if self._fields[name].type == 'selection':
+                res[name] = int(res[name])
+
+        return res
+
+    @api.model
+    def _get_classified_fields(self):
```
Comment (yelizariev) [inline access_apps/models/res_config.py]: Add some comment what and why extension does

**Comment by iledarn** [comment]
@yelizariev, done

File: access_apps/models/res_config.py [hunk 3]
```diff
@@ -26,4 +26,44 @@ def default_get(self, fields):
...
+
+        return res
+
+    @api.multi
+    def execute(self):
+        # base `exectute` doesn't know about new classification - it only has a list of modules to install now
+        res = super(ResConfigSettings, self).execute()
+        # uninstall modules if needed and a user has access
+        to_uninstall = self._get_classified_fields()['to_uninstall']
+        if to_uninstall and self.env['res.users'].has_group('access_apps.grgup_allow_apps_only_from_settings'):
```
Comment (yelizariev) [inline access_apps/models/res_config.py]: grgup_allow_apps_only_from_settings

**Comment by iledarn** [comment]
@yelizariev , done

File: access_apps/security/access_apps_security.xml [hunk 4]
```diff
@@ -8,12 +8,13 @@
...
     <field name="name">Allow install apps only from settings</field>
     <field name="category_id" ref="module_category_access_apps"/>
+    <field name="implied_ids" eval="[(4, ref('base.group_system'))]"/>
   </record>
   <record id="group_allow_apps" model="res.groups">
     <field name="name">Allow install apps</field>
     <field name="category_id" ref="module_category_access_apps"/>
     <field name="users" eval="[(4, ref('base.user_root'))]"/>
-    <field name="implied_ids" eval="[(4, ref('group_allow_apps_only_from_settings')), (4, ref('base.group_system'))]"/>
+    <field name="implied_ids" eval="[(4, ref('base.group_system'))]"/>
```
Comment (yelizariev) [inline access_apps/security/access_apps_security.xml]: If user 
has ``Allow install apps``
but doesn't have ``Allow install apps only from settings``
will he be able to uninstall modules from Settings?

Comment (iledarn) [reply access_apps/security/access_apps_security.xml]: Yes, it's wrong. I faultily removed ``(4, ref('group_allow_apps_only_from_settings'))`` from implied of ``Allow install apps``. Should remove this instead ``(4, ref('base.group_system'))``

Comment (iledarn) [reply access_apps/security/access_apps_security.xml]: The ``Allow install apps`` has full access to ``ir.model.model``, but the xml is wrong anyhow

**Pull request merged**